apps/src/gamelab/GameLab.js

```diff
@@ -313,6 +314,13 @@ GameLab.prototype.setupReduxSubscribers = function (store) {
     var lastState = state;
     state = store.getState();
 
+    if (state.interfaceMode !== lastState.interfaceMode &&
+        state.interfaceMode === GameLabInterfaceMode.ANIMATION &&
+        (!this.studioApp_.hasContainedLevels || (this.studioApp_.hasContainedLevels &&
+        codeStudioLevels.hasValidContainedLevelResult()))) {
```

Comment:
(should also probably double check that my "simplified" logic still gets things right)